### PR TITLE
[DOC] Fixes in doc

### PR DESCRIPTION
**Pull request by MitraDarja**
Some small fixes for the search documentation, some spelling errors, some formatting (table correctly aligned), some missing definitions added and death links corrected.

### CODE REVIEW

File: include/seqan3/search/configuration/all.hpp [hunk 1]
```diff
@@ -52,11 +52,11 @@
...
+ * | **Config**                                                  | **0** | **1** | **2** | **3** | **4** |
+ * | ------------------------------------------------------------|-------|-------|-------|-------|-------|
  * | \ref seqan3::search_cfg::max_error  "0: Max error"          |   ❌   |   ❌   |   ✅   |  ✅    |  ✅    |
  * | \ref seqan3::search_cfg::max_error_rate "1: Max error rate" |   ❌   |   ❌   |   ✅   |  ✅    |  ✅    |
- * | \ref seqan3::search_cfg::output "2: Output"                 |   ✅    |   ✅    |   ❌   |  ✅    |  ✅    |
- * | \ref seqan3::search_cfg::mode "3: Mode"                     |   ✅    |   ✅    |   ✅   |  ❌    |  ✅    |
- * | \ref seqan3::search_cfg::parallel "4: Parallel"             |   ✅    |   ✅    |   ✅   |  ✅    |  ❌    |
+ * | \ref seqan3::search_cfg::output "2: Output"                 |   ✅   |   ✅   |   ❌   |  ✅    |  ✅    |
+ * | \ref seqan3::search_cfg::mode "3: Mode"                     |   ✅   |   ✅   |   ✅   |  ❌    |  ✅    |
+ * | \ref seqan3::search_cfg::parallel "4: Parallel"             |   ✅   |   ✅   |   ✅   |  ✅    |  ❌    |
```
Comment (Irallia) [inline include/seqan3/search/configuration/all.hpp]: If you touch these lines, I would prefer the same amout of empty spaces:
```cpp
 * | \ref seqan3::search_cfg::max_error  "0: Max error"          |   ❌   |   ❌   |   ✅   |   ✅   |   ✅   |
 * | \ref seqan3::search_cfg::max_error_rate "1: Max error rate" |   ❌   |   ❌   |   ✅   |   ✅   |   ✅   |
 * | \ref seqan3::search_cfg::output "2: Output"                 |   ✅   |   ✅   |   ❌   |   ✅   |   ✅   |
 * | \ref seqan3::search_cfg::mode "3: Mode"                     |   ✅   |   ✅   |   ✅   |   ❌   |   ✅   |
 * | \ref seqan3::search_cfg::parallel "4: Parallel"             |   ✅   |   ✅   |   ✅   |   ✅   |   ❌   |
```

File: include/seqan3/search/configuration/max_error.hpp [hunk 2]
```diff
@@ -28,7 +28,8 @@ namespace seqan3::search_cfg
 /*!\brief A configuration element for the maximum number of errors across all error types (mismatches, insertions,
  *        deletions). This is an upper bound of errors independent from error numbers of specific error types.
  * \ingroup search_configuration
- * \details An insertion corresponds to a base inserted into the query that does not occur in the text at the position,
+ * \details A mismatch corresponds to a diverging bases between text and query for a certain position.
```
Comment (Irallia) [inline include/seqan3/search/configuration/max_error.hpp]: without "a"
```cpp
* \details A mismatch corresponds to diverging bases between text and query for a certain position.
```

File: include/seqan3/search/configuration/max_error_rate.hpp [hunk 3]
```diff
@@ -30,7 +30,8 @@ namespace seqan3::search_cfg
  *        (mismatches, insertions, deletions). This is an upper bound of errors independent from error rates of
  *        specific error types.
  * \ingroup search_configuration
- * \details An insertion corresponds to a base inserted into the query that does not occur in the text at the position,
+ * \details A mismatch corresponds to a diverging bases between text and query for a certain position.
```
Comment (Irallia) [inline include/seqan3/search/configuration/max_error_rate.hpp]: same here

File: include/seqan3/search/configuration/detail.hpp [hunk 4]
```diff
@@ -27,13 +27,13 @@ namespace seqan3::detail
  * \details
  *
  * The seqan3::detail::search_config_id used to identify a specific search configuration element independent of
- * it's concrete type and position within the seqan3::search_cfg::search_configuration object.
+ * it's concrete type and position within the seqan3::search_cfg object.
```
Comment (rrahn) [inline include/seqan3/search/configuration/detail.hpp]: Right now, the link points to the namespace, which I fell a little misleading. Instead I would only name the search configuration and link it to the module page of the search configuration. This landing page should then be used to explain more in depth the principles and functionality of the search.

Comment (MitraDarja) [reply include/seqan3/search/configuration/detail.hpp]: I am sorry, I don't get what I should change here.

Comment (eseiler) [inline include/seqan3/search/configuration/detail.hpp]:
```suggestion
 * its concrete type and position within the seqan3::search_cfg object.
```

Comment (rrahn) [reply include/seqan3/search/configuration/detail.hpp]:
```suggestion
 * its concrete type and position within the \ref seqan3::search_cfg::search_configuration "search configuration object".
```

File: include/seqan3/search/configuration/parallel.hpp [hunk 5]
```diff
@@ -22,8 +22,6 @@ namespace seqan3::search_cfg
  *
  * \details
  *
- * With this configuration you can enable the parallel execution of the search algorithm.
```
Comment (smehringer) [inline include/seqan3/search/configuration/parallel.hpp]: why did you remove this?

File: include/seqan3/search/fm_index/fm_index.hpp [hunk 6]
```diff
@@ -48,7 +48,7 @@ class bi_fm_index_cursor;
  *
  * ### Running time / Space consumption
  *
- * \f$SAMPLING\_RATE = 16\f$
+ * \f$SAMPLING\_RATE = 16\f$ \n
```
Comment (smehringer) [inline include/seqan3/search/fm_index/fm_index.hpp]: is this adding a line break in doxygen ? I've never seen this before

File: include/seqan3/search/configuration/parallel.hpp (hunk 5 above)
Comment (MitraDarja) [reply include/seqan3/search/configuration/parallel.hpp]: I have no idea. :flushed:

File: include/seqan3/search/fm_index/fm_index.hpp (hunk 6 above)
Comment (MitraDarja) [reply include/seqan3/search/fm_index/fm_index.hpp]: @smehringer Yes, it puts the sigma on the next line.

Comment (smehringer) [reply include/seqan3/search/fm_index/fm_index.hpp]: Alright
